ui/src/app/pages/homepage/data-access-requirements.tsx

```diff
@@ -0,0 +1,506 @@
+  const {profile} = useStore(profileStore);
+  const [navigate, ] = useNavigation();
+
+  const {module, active} = props;
+  const eligible = true; // TODO
+  const statusTextMaybe = bypassedOrCompletedText(getAccessModuleStatusByName(profile, module));
+
+  // whether this module needs a profile reload
+  const [needsReload, setNeedsReload] = useState(false);
+
+  // whether to show the Two Factor Auth Modal
+  const [showTwoFactorAuthModal, setShowTwoFactorAuthModal] = useState(false);
+
+  const registrationTask = getRegistrationTask(navigate, module);
+  const moduleEnabled = !!registrationTask;
+
+  // kluge until we have fully migrated from the Registration Dashboard:
+  // getRegistrationTask() has onClick() functions for every module, which is generally what we want
+  // but we pop up a modal for Two Factor Auth instead of using the standard task
+  const moduleAction = registrationTask && (module === AccessModule.TWOFACTORAUTH ?
+      () => setShowTwoFactorAuthModal(true) :
+      registrationTask.onClick);
+
+  const Refresh = () => <Button
+      type='primary'
+      style={styles.refreshButton}
+      onClick={() => location.reload()} >
+    <Repeat style={styles.refreshIcon}/> Refresh
+  </Button>;
+
+  const Next = () => <FlexRow style={styles.nextElement}>
+    <span style={styles.nextText}>NEXT</span> <ArrowRight style={styles.nextIcon}/>
+  </FlexRow>;
+
+  const ModuleIcon = () => <div style={styles.moduleIcon}>
+    {cond(
+      // not eligible to complete module
+      [!eligible, () => <MinusCircle data-test-id={`module-${module}-ineligible`} style={{color: colors.disabled}}/>],
+      // eligible and completed or bypassed
+      [eligible && !!statusTextMaybe, () => <CheckCircle data-test-id={`module-${module}-complete`} style={{color: colors.success}}/>],
+      // eligible and incomplete and unbypassed
+      [eligible && !statusTextMaybe, () => <CheckCircle data-test-id={`module-${module}-incomplete`} style={{color: colors.disabled}}/>])}
+  </div>;
+
+  const ModuleBox = ({children}) => {
+    return active ?
+        <Link onClick={() => {
+          setNeedsReload(true);
+          moduleAction();
+        }}>
+          <FlexRow style={styles.activeModuleBox}>{children}</FlexRow>
+        </Link> :
+        <FlexRow style={styles.moduleBox}>{children}</FlexRow>;
+  };
+
+  const Module = () => <FlexRow data-test-id={`module-${module}`}>
+    <FlexRow style={styles.moduleCTA}>
+      {active && (needsReload ? <Refresh/> : <Next/>)}
+    </FlexRow>
+    <ModuleBox>
+      <ModuleIcon/>
+      <FlexColumn>
+        <div style={active ? styles.activeModuleText : styles.inactiveModuleText}>
+          {moduleLabels.get(module)}
+        </div>
+        {statusTextMaybe && <div style={styles.moduleDate}>{statusTextMaybe}</div>}
+      </FlexColumn>
+    </ModuleBox>
+    {showTwoFactorAuthModal && <TwoFactorAuthModal
+        onClick={() => setShowTwoFactorAuthModal(false)}
```

Comment:
DAR has simpler refresh logic than registration-dashboard, so the latter has additional logic here